lib/network/modules/components/nodes/util/ShapeBase.js

```diff
@@ -43,16 +43,23 @@ class ShapeBase extends NodeBase {
    * @param {boolean} selected
    * @param {boolean} hover
    * @param {ArrowOptions} values
+   * @param {function} customRenderer - a custom shape renderer similar to getShape(shape) functions
    * @private
    */
-  _drawShape(ctx, shape, sizeMultiplier, x, y, selected, hover, values) {
+  _drawShape(ctx, shape, sizeMultiplier, x, y, selected, hover, values, customRenderer) {
     this.resize(ctx, selected, hover, values);
     this.left = x - this.width / 2;
     this.top = y - this.height / 2;
 
     this.initContextForDraw(ctx, values);
```

Comment:
I wouldn't run this for custom shapes. I think this is tightly coupled to built-in shapes and could cause problems later on (we couldn't change anything here as it would break people's custom shapes).

Reply:
I'm not sure what you mean. Do you think there should be a different function for the custom shape draw?

Reply:
No, I mean specifically the call to `initContextForDraw`.

Reply:
Ok so I think the solution is to move `initContextForDraw` to line 61 in the case the shape is not 'custom'

Reply:
Yeah, maybe it would've been clearer if I said something like “move `initContextForDraw` to the else block”.